### ISSUE TITLE
[Troubleshoot]: π0.5 PPO training on LIBERO-Spatial not converging as expected

### ISSUE THREAD

**Issue by zzzyzh** [RLinf/RLinf#585]
### Problem description

Hi, thank you very much for your excellent work and for releasing the codebase.

I came across the training curves you shared for OpenPI 0.5 on LIBERO Spatial tasks in this GitHub [issue:348](https://github.com/RLinf/RLinf/issues/348)

From the figures, it appears that with π0.5 + PPO, the model performance reaches around 0.9 success rate within ~300 steps, and even π0 seems to reach 0.9+ within ~100 steps.

However, in my reproduction experiments under the same task setting (LIBERO Spatial with PPO), I observe noticeably slower convergence. Specifically, at around 300 training steps, the success rate of π0.5 is only about 0.8, which is significantly lower than what is reported in the shared curves.

<img width="468" height="438" alt="Image" src="https://github.com/user-attachments/assets/cffa04dd-7874-47dc-bc91-6b9147b87e9d" />

I would like to ask:
1. Are there any non-obvious hyperparameters, implementation details, or training tricks that are particularly important for achieving the reported convergence?
2. Is the curve in the issue averaged over multiple seeds, or does it correspond to a single run?
3. Are there any differences in environment versions, observation preprocessing, reward normalization, or PPO-specific settings that could significantly affect early-stage convergence?

Any clarification or guidance would be greatly appreciated.
Thank you again for your great work and for making the project publicly available.

### Configuration YAML file

YAML config file:

You can also paste the full config here.
```
defaults:
  - env/libero_spatial@env.train
  - env/libero_spatial@env.eval
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: all

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "libero_spatial_ppo_openpi_pi05"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1

  only_eval: False
  eval_policy_path: null # Optional: .pt file or None, if None, will use the checkpoint in rollout.model.model_path
  val_check_interval: -1
  save_interval: 40

  resume_dir: null

algorithm:
  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 1
  reward_coef: 1.0

  rollout_epoch: 8
  eval_rollout_epoch: 1 # set eval_rollout_epoch > 0 when enable runner.only_eval or runner.val_check_interval > 0

  reward_type: chunk_level
  logprob_type: chunk_level
  entropy_type: token_level
  update_epoch: 1

  adv_type: gae
  loss_type: actor_critic
  loss_agg_func: "token-mean" 
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  filter_rewards: False
  rewards_lower_bound: 0.1
  rewards_upper_bound: 0.9
  # params for generation
  sampling_params:
    do_sample: True
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 50
    top_p: 1.0
    repetition_penalty: 1.0
    add_BOS: False

  # length argument for autoregressive sampling
  # max length means max amount of tokens to generate
  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"
  enable_offload: False

  # Override the default values in env/libero_spatial
  train:
    total_num_envs: 64
    max_episode_steps: 240 # max episode steps for truncation
    max_steps_per_rollout_epoch: 240
  eval:
    total_num_envs: 500
    auto_reset: True
    ignore_terminations: True
    max_episode_steps: 240
    max_steps_per_rollout_epoch: 240
    group_size: 1
    is_eval: True
    video_cfg:
      save_video: True
      video_base_dir: ${runner.logger.log_path}/video/eval
      save_interval: 20  # Save video every N rollouts (default: 1, save every rollout)

rollout:
  group_name: "RolloutGroup"
  generation_backend: "huggingface"
  recompute_logprobs: False
  unnorm_key: libero_10
  enable_offload: False
  pipeline_stage_num: 1

  model:
    model_path: "/mnt/cpfs/yanzhonghao/ckpts/RLinf-Pi05-SFT"
    precision: ${actor.model.precision}

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 128
  global_batch_size: 2048 # maximum : global batch size = micro batch size * actor num_processes.
  seed: 42
  enable_offload: False

  # Override the default values in model/pi0_5
  model:
    model_path: "/mnt/cpfs/yanzhonghao/ckpts/RLinf-Pi05-SFT"
    model_type: "openpi"
    num_action_chunks: 5 # interface for the env
    num_steps: 3
    add_value_head: True
    # openpi specific parameters
    openpi:
      value_after_vlm: True

  optim:
    lr: 5.0e-6
    value_lr: 1.0e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-05
    weight_decay: 0.01
    clip_grad: 1.0
    critic_warmup_steps: 0

  # Override the default values in training_backend/fsdp
  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "no_shard"
    gradient_checkpointing: False # for openpi, gradient checkpointing is not supported, please do not change this value
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

reward:
  use_reward_model: False

critic:
  use_critic_model: False
```

### Log file

You can find the log file in `logs/` folder or the $output_dir/$experiment_name folder (defined in the yaml config) if you are using our example scripts.

Log file:

If you cannot find the log, please provide the full log messages here.
```
<paste log here>
```

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by chenkang455** [comment]
@yfpeng1234 Thanks for providing it. We are reproducing this problem.

**Comment by chenkang455** [comment]
@zzzyzh From our experiments, it seems that the main branch has some bugs while the v01 version is working nicely. We are debugging it; thank you for your report! 

Check v01: https://github.com/RLinf/RLinf/tree/release/v0.1

Additionally, the results reported in our paper are based on the evaluation results, not the rollout depicted here. Eval results will be higher than the rollout performance since the noise injection during the training stage will deteriorate the performance.

<img width="1254" height="714" alt="Image" src="https://github.com/user-attachments/assets/4da5d5b0-59e4-4588-8329-395950192e85" />

**Comment by zzzyzh** [comment]
Thank you very much for the quick reproduction and for pinpointing the issue in the main branch. 
I will switch to v0.1 for now and look forward to re-running the experiments once the pi0.5 code in the main branch is fixed.

**Comment by chenkang455** [comment]
@zzzyzh Thanks for your feedback! Besides, I recommend that you run ``libero_spatial_openpi`` for a fast try, which is expected to converge in 100 steps.

Below is the training curve for ``pi05 libero-spatial``:

<img width="856" height="375" alt="Image" src="https://github.com/user-attachments/assets/dee400e3-b1d9-41a8-b981-598ffaf5e4c9" />

Below is the training curve for ``pi0 libero-spatial``:

<img width="883" height="398" alt="Image" src="https://github.com/user-attachments/assets/35d28962-321a-4f3a-8112-9373bf7ed124" />

**Comment by zzzyzh** [comment]
I also noticed that some hyperparameters used for pi0 in the pi-RL paper do not seem to be fully aligned with those provided in the repository, for example the learning rate.
In addition, I observed that the entropy bonus is set to 0 across the configurations. Could you clarify the rationale behind this choice?
<img width="1096" height="736" alt="Image" src="https://github.com/user-attachments/assets/58a5a755-c970-45f6-9e01-6a33b3ffd5a8" />

**Comment by chenkang455** [comment]
@zzzyzh As for the first question, the reason behind it is that the config provided by our repo is slightly different from our paper version. We have rectified the new config in our paper, which is estimated to be updated soon: 

<img width="556" height="295" alt="Image" src="https://github.com/user-attachments/assets/82d989dd-c549-41c7-8248-fda38ba4d140" />

Besides, the entropy bonus is designed for the flow-noise method, which is not the default yaml for libero_spatial_openpi (flow-sde). You can find the guidance here: https://rlinf.readthedocs.io/en/latest/rst_source/examples/pi0.html#:~:text=2.2%20Algorithm%20Configuration.

**Comment by chenkang455** [comment]
@zzzyzh We have identified a potential bug, and the modified PR is available here: https://github.com/RLinf/RLinf/pull/623. This bug is originated from the version after v01.

Therefore, if you are using versions between them, please consider modifying the following parameters:
```
adam_eps: 1.0e-05 -> 1.0e-08
```
